src/loaders/compute_tools/tool_common.py

```diff
@@ -0,0 +1,458 @@
+            '--source_data_dir', required=True, type=str,
+            help='Source data (e.g. genome files) directory. '
+                + '(e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207'
+        )
+
+        # Optional arguments
+        optional.add_argument(
+            f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', type=str,
+            default=loader_common_names.DEFAULT_KBASE_COLL_NAME,
+            help=loader_common_names.KBASE_COLLECTION_DESCR
+        )
+        optional.add_argument(
+            '--root_dir', type=str, default=loader_common_names.ROOT_DIR, help='Root directory.'
+        )
+        optional.add_argument(
+            '--threads', type=int,
+            help='Total number of threads used by the script. (default: half of system cpu count)'
+        )
+        optional.add_argument(
+            '--program_threads', type=int, default=32,
+            help='Number of threads to execute a single tool command. '
+                + 'threads / program_threads determines the number of batches. (default: 32)'
+        )
+        optional.add_argument(
+            '--node_id', type=str, default=str(uuid.uuid4()), help='node ID for running job'
+        )
+        optional.add_argument('--debug', action='store_true', help='Debug mode.')
+        optional.add_argument(
+            '--data_id_file', type=argparse.FileType('r'),
+            help="tab separated file containing data ids for the running job "
+                + f"(requires '{DATA_ID_COLUMN_HEADER}' as the column name)"
+        )
+        optional.add_argument(
+            '--source_file_ext', type=str, default='.fa',
+            help='Select files from source data directory that match the given extension.'
+        )
+        return parser.parse_args()
+
+    def _get_data_ids(self):
+        # get all data IDs (folder name) from source data directory
+        all_data_ids = [path for path in os.listdir(self._source_data_dir) if
+                        os.path.isdir(os.path.join(self._source_data_dir, path))]
+
+        if not all_data_ids:
+            raise ValueError('Please download genome files first.')
+
+        # get data ids for the running job
+        if self._data_id_file:
+            # parse data ids from tab separated file
+            with self._data_id_file:
+                df = pd.read_csv(self._data_id_file, sep='\t')
+                try:
+                    data_ids = df[DATA_ID_COLUMN_HEADER]
+                except KeyError:
+                    raise ValueError(
+                        f"Please ensure {DATA_ID_COLUMN_HEADER} column exists in the "
+                        + "data id file.")
+
+            if not set(all_data_ids) >= set(data_ids):
+                raise ValueError(
+                    f'Data files for {set(data_ids) - set(all_data_ids)} are not downloaded yet')
+        else:
+            # executing all data IDs
+            data_ids = all_data_ids
+        return data_ids
```

Comment:
maybe we should return `set(data_ids)`? It might be possible if the data_id_file contains duplicate data_ids.

Reply:
Sure, although `task_generator.py` is creating all the lists at the moment so it seems unlikely. I'll do it in a future PR.